lib/matplotlib/tests/test_text.py

```diff
@@ -393,3 +393,14 @@ def test_text_stale():
     assert not ax1.stale
     assert not ax2.stale
     assert not fig.stale
+
+
+@image_comparison(baseline_images=['agg_text_clip'],
+                  extensions=['png'])
+def test_agg_text_clip():
+    np.random.seed(1)
+    fig, (ax1, ax2) = plt.subplots(2)
+    for x, y in np.random.rand(10, 2):
+        ax1.text(x, y, "foo", clip_on=True)
+        ax2.text(x, y, "foo")
```

Comment:
I would have gone with different text for each axes, just to be sure it's clipped to the right place.